qa/rolling-upgrade/src/test/java/org/opensearch/upgrades/IndexingIT.java

```diff
@@ -106,14 +106,25 @@ private void waitForSearchableDocs(String index, int shardCount, int replicaCoun
             Request segrepStatsRequest = new Request("GET", "/_cat/segments/" + index + "?s=shard,segment,primaryOrReplica");
             segrepStatsRequest.addParameter("h", "index,shard,primaryOrReplica,segment,docs.count");
             Response segrepStatsResponse = client().performRequest(segrepStatsRequest);
-            logger.info("--> _cat/segments response\n {}", EntityUtils.toString(segrepStatsResponse.getEntity()));
             List<String> responseList = Streams.readAllLines(segrepStatsResponse.getEntity().getContent());
-            for (int segmentsIndex=0; segmentsIndex < responseList.size();) {
-                String[] primaryRow = responseList.get(segmentsIndex++).split(" +");
+            logger.info("--> _cat/segments response\n {}", responseList.toString().replace(',', '\n'));
```

Comment:
Rather than parsing cat segments at all, should we simplify by enumerating the replicas and assert doc count?

Reply:
Thanks @mch2 for the feedback. We actually started with asserting doc counts on replicas. But, later on changed  logic to assert on segment level doc counts. This was done for couple of reasons captured in PR https://github.com/opensearch-project/OpenSearch/pull/8267 (copy pasting below for visibility). 
1. In order to identify replica and nodes where those shards were sitting, we needed to perform extra API {index}/_stats which contain shard allocations. This was followed by targeted search on individual nodes.
2. _cat/segments was identified better alternative as there was no need to build shard <-> node allocation map and though it verifies segment level document count., which is more granular (though not required).